[CI] SearchQueryThenFetchAsyncActionTests testMinimumVersionBetweenNewAndOldVersion failing

**Build Scans:**
- [elasticsearch-periodic #5069 / single-processor-node-tests](https://gradle-enterprise.elastic.co/s/3hhpt3wfyj3sw)
- [elasticsearch-periodic-platform-support #5047 / oraclelinux-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/ylioiiouw4qks)
- [elasticsearch-pull-request #43431 / part-1](https://gradle-enterprise.elastic.co/s/5fz6wtssce7na)
- [elasticsearch-periodic-platform-support #5041 / rocky-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/62zsczuax56bo)
- [elasticsearch-periodic-platform-support #5041 / rhel-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/zeeosmzgsg3d6)
- [elasticsearch-periodic-platform-support #5023 / rhel-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/cr3tpg6a75rgy)
- [elasticsearch-periodic-platform-support #5017 / ubuntu-2004_platform-support-unix](https://gradle-enterprise.elastic.co/s/p66ejxachclla)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.action.search.SearchQueryThenFetchAsyncActionTests.testMinimumVersionBetweenNewAndOldVersion" -Dtests.seed=C20A79FFF301F056 -Dtests.configure_test_clusters_with_one_processor=true -Dtests.locale=vmw -Dtests.timezone=America/Grand_Turk -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.action.search.SearchQueryThenFetchAsyncActionTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testMinimumVersionBetweenNewAndOldVersion),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: an instance of org.elasticsearch.action.search.VersionMismatchException
     but: <java.lang.NullPointerException: Cannot invoke "java.util.function.BiFunction.apply(Object, Object)" because "this.responseWrapper" is null> is a java.lang.NullPointerException
```

**Issue Reasons:**
- [8.x] 2 consecutive failures in test testMinimumVersionBetweenNewAndOldVersion
- [8.x] 7 failures in test testMinimumVersionBetweenNewAndOldVersion (3.0% fail rate in 233 executions)
- [8.x] 2 failures in step rhel-8_platform-support-unix (40.0% fail rate in 5 executions)
- [8.x] 4 failures in pipeline elasticsearch-periodic-platform-support (66.7% fail rate in 6 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -247,9 +247,6 @@ tests:
 - class: org.elasticsearch.search.basic.SearchWithRandomDisconnectsIT
   method: testSearchWithRandomDisconnects
   issue: https://github.com/elastic/elasticsearch/issues/116175
-- class: org.elasticsearch.action.search.SearchQueryThenFetchAsyncActionTests
-  method: testMinimumVersionBetweenNewAndOldVersion
-  issue: https://github.com/elastic/elasticsearch/issues/117485
 - class: org.elasticsearch.discovery.ClusterDisruptionIT
   method: testAckedIndexing
   issue: https://github.com/elastic/elasticsearch/issues/117024

--- a/server/src/test/java/org/elasticsearch/action/search/SearchQueryThenFetchAsyncActionTests.java
+++ b/server/src/test/java/org/elasticsearch/action/search/SearchQueryThenFetchAsyncActionTests.java
@@ -272,16 +272,20 @@ public class SearchQueryThenFetchAsyncActionTests extends ESTestCase {
     }
 
     public void testMinimumVersionBetweenNewAndOldVersion() {
-        var oldVersion = new VersionInformation(
-            VersionUtils.getFirstVersion(),
-            IndexVersions.MINIMUM_COMPATIBLE,
-            IndexVersionUtils.randomCompatibleVersion(random())
-        );
-
         var newVersion = new VersionInformation(
             VersionUtils.maxCompatibleVersion(VersionUtils.getFirstVersion()),
             IndexVersions.MINIMUM_COMPATIBLE,
             IndexVersion.current()
+        );
+
+        var oldVersion = new VersionInformation(
+            VersionUtils.randomVersionBetween(
+                random(),
+                Version.CURRENT.minimumCompatibilityVersion(),
+                VersionUtils.getPreviousVersion(newVersion.nodeVersion())
+            ),
+            IndexVersions.MINIMUM_COMPATIBLE,
+            IndexVersionUtils.randomCompatibleVersion(random())
         );
 
         var minVersion = VersionUtils.randomVersionBetween(

